graphql_compiler/macros/macro_edge/helpers.py

```diff
@@ -53,6 +60,137 @@ def get_directives_for_ast(ast):
+    # Schema validation has ensured this exists
+    current_name = tag_directive.arguments[0].value.value
+    new_name = name_change_map[current_name]
+    renamed_tag_directive = copy(tag_directive)
+    renamed_tag_directive.arguments = [Argument(Name('tag_name'), StringValue(new_name))]
+    return renamed_tag_directive, new_name != current_name
+
+
+def _replace_tag_names_in_filter_directive(name_change_map, filter_directive):
+    """Return the new directive, and whether it is different from the old."""
+    made_changes = False
+    filter_with_renamed_args = copy(filter_directive)
+    filter_with_renamed_args.arguments = []
+    for argument in filter_directive.arguments:
+        if argument.name.value == 'op_name':
+            filter_with_renamed_args.arguments.append(argument)
+        elif argument.name.value == 'value':
+            new_value_list = []
+            for value in argument.value.values:
+                if is_tag_argument(value.value):
+                    current_name = get_directive_argument_name(value.value)
+                    new_name = name_change_map[current_name]
+                    if new_name != current_name:
+                        made_changes = True
+                    new_value_list.append(StringValue('%' + new_name))
+                elif is_variable_argument(value.value):
+                    new_value_list.append(value)
+                else:
+                    raise AssertionError(u'Value was neither tag nor filter: {} {}'
+                                         .format(filter_directive, value.value))
+            filter_with_renamed_args.arguments.append(
+                Argument(Name('value'), value=ListValue(new_value_list)))
+        else:
+            raise AssertionError(u'Unknown argument name {} in filter'
+                                 .format(argument.name.value))
+    return filter_with_renamed_args, made_changes
+
+
+def _replace_tag_names_at_current_node(name_change_map, ast):
+    """Replace tag names that are already in use at the root of the AST."""
+    # Rename tag names in @tag and @filter directives, and record if we made changes
+    made_changes = False
+    new_directives = []
+    for directive in ast.directives:
+        if directive.name.value == TagDirective.name:
+            renamed_tag_directive, made_changes_to_tag = _replace_tag_names_in_tag_directive(
+                name_change_map, directive)
+            made_changes = made_changes or made_changes_to_tag
+            new_directives.append(renamed_tag_directive)
+        elif directive.name.value == FilterDirective.name:
+            filter_with_renamed_args, made_changes_to_filter = (
+                _replace_tag_names_in_filter_directive(name_change_map, directive))
+            made_changes = made_changes or made_changes_to_filter
+            new_directives.append(filter_with_renamed_args)
+        else:
+            new_directives.append(directive)
+    return made_changes, new_directives
```

Comment:
Let's keep the API consistent across functions. This function returns `(made_changes, stuff)` but others (e.g. on line 113) return `(stuff, made_changes)`.